pkg/cosign/verify.go

```diff
@@ -677,6 +680,11 @@ func verifyInternal(ctx context.Context, sig oci.Signature, h v1.Hash,
 		return false, fmt.Errorf("unable to verify bundle: %w", err)
 	}
 
+	// If the --offline flag was specified, fail here
+	if !bundleVerified && co.Offline {
```

Comment:
Right now `VerifyBundle` checks if there's a bundle AND verifies the bundle, and right now it seems like `err` is returned if there is an error checking, but `nil` error can also be returned when there is NO bundle. 

Maybe it can be changed to:
* if hasBundle(sig) -> verifyBundle -> if err, return err (always fail if the provided bundle is bad)
* if offline -> return now, offline verification failed
* check online

@znewman01 make sense?

Reply:
+1 to this flow!

Reply:
sgtm, updated

Reply:
wait, is it updated? I don't see it

Reply:
I think it is now. I hate that `VerifyBundle` has the double failure meaning (no bundle vs bad bundle), but it checks out now.

If there's no bundle we'll return false, nil.